app/screens/receive-bitcoin-screen/receive-usd.tsx

```diff
@@ -145,6 +146,45 @@ const ReceiveUsd = () => {
   const { lnUpdate } = useSubscriptionUpdates()
   const [showMemoInput, setShowMemoInput] = useState(false)
   const [showAmountInput, setShowAmountInput] = useState(false)
+  const { timeLeft, startCountdownTimer, resetCountdownTimer } = useCountdownTimer()
+  const timeUntilInvoiceExpires = useMemo(() => {
```

Comment:
I think ```timeUntilInvoiceExpires``` is a misleading name, because on subsequent renders it is no longer actually the time until invoice expires. You can get rid of lines 150-158 all together and add that logic to the effect in line 179 while changing line 187 to just depend on the invoice.

Reply:
Yeah I agree and I have made the change.